lib/mapObjects/CObjectHandler.cpp

```diff
@@ -215,6 +216,8 @@ void CGObjectInstance::initObj(CRandomGenerator & rand)
 		blockVisit = true;
 		break;
 	}
+
+	VLC->objtypeh->getHandlerFor(ID, subID)->setAmbient(this);
```

Comment:
too much code duplication, if it too difficult to ensure CGObjectInstance::initObj call for all descendants move this initialization to CGameState::initMapObjects()